src/browser/pages/vscode.ts

```diff
@@ -5,18 +5,54 @@ const options = getOptions()
 // TODO: Add proper types.
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
-let nlsConfig: any
+export const nlsConfigElementId = "vscode-remote-nls-configuration"
+
+type NlsConfiguration = {
+  locale: string
+  availableLanguages: { [key: string]: string } | {}
+  _languagePackId?: string
+  _translationsConfigFile?: string
+  _cacheRoot?: string
+  _resolvedLanguagePackCoreLocation?: string
+  _corruptedFile?: string
+  _languagePackSupport?: boolean
+  loadBundle?: any
+}
+
+/**
+ * A helper function to get the NLS Configuration settings.
+ *
+ * This is used by VSCode for localizations (i.e. changing
+ * the display language).
+ **/
+export function getNlsConfiguration(document: Document) {
+  const nlsConfigElement = document?.getElementById(nlsConfigElementId)
+  const nlsConfig = nlsConfigElement?.getAttribute("data-settings")
+
+  if (!document || !nlsConfigElement || !nlsConfig) {
+    return undefined
+  }
```

Comment:
The downside of this approach is:
- we don't log any errors
- we don't know which piece of this came back false/`undefined`

Would it be worth it to break this up and use a `console.error` for each? Not sure how often people experience errors with the NLS Configuration.

Reply:
IMO we should throw individual errors for each one, then wherever we call this we can catch and do a single `console.error`.

Reply:
something like?
```javascript
try {
  if (!document) {
    throw Error("no document)
  }
  
  // etc..
} catch (e) {
  console.error("something broke", e)
}
```

genius! I'll refactor

Reply:
> wherever we call this we can catch and do a single

Oh, don't use `try/catch` inside. Just wherever we call it. Got it!